tests/acceptance/resources/keywords.resource

```diff
@@ -1,88 +1,90 @@
-    Wait Until Keyword Succeeds    20 seconds    0.5 second
-    ...    Verify URL Is Reachable    /api/v1/health
-
-Stop Rfhub2
-    [Documentation]
-    ...    Stops the rfhub2 process created by "Start Rfhub2"
-    Terminate Process    ${rfhub2_process}
-    ${result}    Get Process Result
-    Run Keyword If    len('''${result.stderr}''') > 0
-    ...    Log    rfhub2 stderr: ${result.stderr}    DEBUG
-    
-Open Browser With App On Mainpage
-    Open Browser    url=localhost:8000    browser=gc
-    Maximize Browser Window
-
-Navigate To Main Page
-    Go To    localhost:8000
-
-Collections Count On Main Page Should Be ${n}
-    Reload Page
-    Element Text Should Be    ${main_page_collection_number}    Collections (${n})
-
-Start Cli Package With Options
-    [Arguments]    ${options}
-    [Documentation]
-    ...    Attempt to start the rfhub2-cli package with the given options
-    ...    The stdout of the process will be in a test suite
-    ...    variable named \${output}
-    ${output}    Run    rfhub2-cli ${options}
-    Set Suite Variable    ${output}
-
-Start Cli Package
-    [Documentation]
-    ...    Attempt to start the rfhub2-cli package with the default options
-    ...    The stdout of the process will be in a test suite
-    ...    variable named \${output}
-    ${output}    Run    rfhub2-cli
-    Set Suite Variable    ${output}
-
-Output Should Contain
-    [Arguments]    @{patterns}
-    [Documentation]
-    ...    Fail if the output from the previous command doesn't contain the given string
-    ...    This keyword assumes the output of the command is in
-    ...    a test suite variable named \${output}
-    ...    Note: the help will be automatically wrapped, so
-    ...    you can only search for relatively short strings.
-    FOR    ${pattern}    IN    @{patterns}
-        Run keyword if    '''${pattern}''' not in '''${output}'''
-        ...    Fail    Output did not contain '${pattern}'
-    END
-
-Verify URL Is Reachable
-    # This could be useful in more places than just API tests.
-    # Maybe it should be moved to a different file...
-    [Arguments]    ${URL}    ${PORT}=8000
-    [Documentation]    Fail if the given URL doesn't return a status code of 200.
-    Create Session    tmp    http://localhost:${PORT}
-    ${response}    Get Request    tmp    ${url}
-    Should Be Equal As Integers    ${response.status_code}    200
-
-${endpoint} Endpoint Should Have ${n} Items
-    Create Session    tmp    http://localhost:8000/api/v1
-    ${response}    Get Request    tmp    ${endpoint}/
-    Should Be Equal As Integers    ${response.status_code}    200
-    ${response_content}    Set Variable    ${response.json()}
-    Length Should Be    ${response_content}    ${n}
-    
-
-
+*** Settings ***
+Library    SeleniumLibrary
+Library    OperatingSystem
+Library    Process
+Library    RequestsLibrary
+Library    Collections
+
+*** Variables ***
+${activate_venv}    . atests-venv/bin/activate &&
```

Comment:
This should be more customizable to allow running without the separate venv. Do you have an idea how to manage this? Maybe something like what we had for switching between sqlite/postgres db for tests in old repository?

Reply:
@MaciejWiczk Yes, it is making process running rfhub2 webserver and rfhub2 cli in different virtualenv than the rest of the CI pipeline and effectively checking if pip installation works properly.
I think it could be handled similarly as DB_URI in one of last PRs. So by default it would be defined as empty, and overriden by config file value. It would allow for running the tested application in current env/virtualenv in development if required and in isolated virtualenv in CI.
Would you like to implement it?
PS. This setup would also allow testing different installation modes in separate virtualenvs (SQLite-only, postgres), but I do not think that it is worth the effort for now. But still it is something worth to remember for the future or other projects.

Reply:
I will take it on. I'm thinking that later on it might be changed into env variable if we would like to use build matrix with python 3.7 and 3.8